server/storage/mvcc/kvstore_txn.go

```diff
@@ -120,9 +126,20 @@ func (tr *storeTxnRead) rangeKeys(ctx context.Context, key, end []byte, curRev i
 				zap.Error(err),
 			)
 		}
+		totalBytes += int64(len(vs[0]))
+		count += 1
+		if ro.MaxBytes > 0 && totalBytes >= ro.MaxBytes {
+			break
+		}
```

Comment:
Do this check before `Unmarshal` (line 123)

Reply:
Done